BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/Skonto/Models/SkontoViewModel.swift

```diff
@@ -151,23 +158,52 @@ class SkontoViewModel {
     }
 
     func setSkontoAmountToPayPrice(_ price: String) {
-        guard let price = convertPriceStringToPrice(price: price),
-              price.value <= amountToPay.value else {
+        setPrice(
+            price,
+            maxValue: amountToPay.value,
+            errorMessage: NSLocalizedStringPreferredGiniBankFormat("ginibank.skonto.withdiscount.validation",
+                                                                   comment: "Discounted value cannot exceed initial...")
+        ) { validatedPrice in
+            skontoAmountToPay = validatedPrice
+            updateDocumentPagesModelData()
+            recalculateSkontoPercentage()
+        }
+    }
+
+    func setAmountToPayPrice(_ price: String) {
+        setPrice(
+            price,
+            maxValue: maximumAmountToPayValue,
+            errorMessage: NSLocalizedStringPreferredGiniBankFormat("ginibank.skonto.withoutdiscount.validation",
```

Comment:
[suggestion] I think for better readability is better to have a separate let for the `errorMessage` value and to fix the indentation for the `setPrice` method. What do you say?